### PR TITLE
Use go 1.5 on the build machine

### LINKED ISSUE #937: Support Go 1.5
There are many benefits to using Go 1.5, including the new pause-less GC. However, we have run into several issues using Go 1.5 to build the registry, both at compile time and runtime. Let's collect them here and burn down towards getting there for the next registry release.

- [x] #887
- [ ] #910
- [x] #913

cc @RichardScothern

### PR DESCRIPTION
**Pull request by RichardScothern**
Closes #937 

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -115,8 +121,6 @@ test:
   # Aggregate and report to coveralls
     - gvm use stable; go list ./... | xargs -L 1 -I{} cat "$GOPATH/src/{}/coverage.out" | grep -v "$CIRCLE_PAIN" >> ~/goverage.report:
```
Comment (dmp42) [inline circle.yml]: Shouldn't we specify the build tags here as well?
Admittedly we don't do anything with the coverage report but then...

File: circle.yml [hunk 2]
```diff
@@ -73,40 +71,48 @@ test:
...
-    #     pwd: $BASE_BLEED
+  override:
+     - gvm use bleed; go test -test.v -test.short ./...:
+         timeout: 600
+         pwd: $BASE_BLEED
 
   # Test stable, and report
   # Preset the goverall report file
-    - echo "$CIRCLE_PAIN" > ~/goverage.report
+  # - echo "$CIRCLE_PAIN" > ~/goverage.report
```
Comment (dmp42) [inline circle.yml]: You need to move that commented out section upper. It broke circle (questionable) parsing.

**Comment by stevvooe** [comment]
@RichardScothern Shouldn't this just update to use go 1.5? Why do a dual build?

We should only every support one version of Go -- the latest one.

File: circle.yml (hunk 1 above)
Comment (RichardScothern) [inline circle.yml]: done

**Comment by RichardScothern** [comment]
updated in anticipation of switching to go 1.5. ping @stevvooe @dmp42

**Comment by dmp42** [comment]
@stevvooe testing both the version we run for *and* tip all the time might help in catching up future problems earlier and make upgrades easier in the future - agreed, it will make CI slower.

wdyt?

**Comment by dmp42** [comment]
LGTM otherwise

**Comment by stevvooe** [comment]
@dmp42 That does have its benefits, but I'd really only like us to have one "supported Go version" at any time. I don't want to give us the option of lagging behind.

**Comment by stevvooe** [comment]
LGTM

Merge at your leisure.

**Pull request merged**